src/gluonts/model/rotbaum/_preprocess.py

```diff
@@ -306,6 +320,15 @@ def _pre_transform(cls, time_series_window) -> Tuple:
             },
         )
 
+    def create_cat_feats(self, feat_list: List) -> List:
```

Comment:
Can you explain what this does? Looking at the code, it's not intuitive to me.

Reply:
It is just a helper function to make one hot encodings. I can change it's name to that and make it a private function now that you mention it since it should not be called from outside the class.

Reply:
Looking at the signature, it's not entirely clear that it does one hot-encoding. I would either state this in a doc-string or the method name.

Personally, I don't care too much about making it "private" or not.

Reply:
I think the code can be improved if you use `zip`:

```python
def one_hot(feat_list, static_cardinality):
    xs = [0] * sum(static_cardinality)

    current = 0

    for feat, cardinality in zip(feat_list, static_cardinality):
        idx = cardinality - feat - 1
        xs[current + idx] = 1
        current += cardinality

    return xs


def one_hot2(feat_list, static_cardinality):
    res = []

    for feat, cardinality in zip(feat_list, static_cardinality):
        idx = cardinality - feat - 1
        one_hot = [0] * cardinality
        one_hot[idx] = 1
        res.extend(one_hot)

    return res
```

I think it would make sense to add comments to how this method works.

Also, you version doesn't assign things in order.